tests/sentry/api/test_scim.py

```diff
@@ -0,0 +1,295 @@
+import pytest
+from django.urls import reverse
+
+from sentry.models import AuthProvider, InviteStatus, OrganizationMember
+from sentry.testutils import APITestCase
+
+CREATE_USER_POST_DATA = {
+    "schemas": ["urn:ietf:params:scim:schemas:core:2.0:User"],
+    "userName": "test.user@okta.local",
+    "name": {"givenName": "Test", "familyName": "User"},
+    "emails": [{"primary": True, "value": "test.user@okta.local", "type": "work"}],
+    "displayName": "Test User",
+    "locale": "en-US",
+    "externalId": "00ujl29u0le5T6Aj10h7",
+    "groups": [],
+    "password": "1mz050nq",
+    "active": True,
+}
+
+
+class SCIMUserTestsPermissions(APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+
+    def test_cant_use_scim(self):
+        url = reverse("sentry-scim-organization-members-index", args=[self.organization.slug])
+        response = self.client.get(url)
+        assert response.status_code == 403
+
+    def test_cant_use_scim_even_with_authprovider(self):
+        AuthProvider.objects.create(organization=self.organization, provider="dummy")
+        url = reverse("sentry-scim-organization-members-index", args=[self.organization.slug])
+        response = self.client.get(url)
+        assert response.status_code == 403
+
+
+class SCIMUserTests(APITestCase):
+    def setUp(self):
+        super().setUp()
+        auth_provider = AuthProvider.objects.create(
+            organization=self.organization, provider="dummy"
+        )
+        with self.feature({"organizations:sso-scim": True}):
+            auth_provider.enable_scim(self.user)
+            auth_provider.save()
+        self.login_as(user=self.user)
+
+    def test_user_flow(self):
```

Comment:
i decided to have a bunch of assertions with this one test that does a lot -- it seemed more elegant to test everything this way as i could use the scim routes to do the setup and teardown of the users. if this seems ugly to others i could change to split into multiple tests